src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -1728,8 +1728,21 @@ private void ReportMethodSignatureCollision(DiagnosticBag diagnostics, SourceMem
             var methodName = (method1.MethodKind == MethodKind.Destructor && method2.MethodKind == MethodKind.Destructor) ?
                 "~" + this.Name :
                 method1.Name;
-            // Type '{1}' already defines a member called '{0}' with the same parameter types
-            diagnostics.Add(ErrorCode.ERR_MemberAlreadyExists, method1.Locations[0], methodName, this);
+            if (methodName == WellKnownMemberNames.InstanceConstructorName)
+            {
+                // Type '{0}' already defines a constructor with the same parameter types
+                diagnostics.Add(ErrorCode.ERR_ConstructorAlreadyExists, method1.Locations[0], this);
+            }
+            else if (methodName == WellKnownMemberNames.StaticConstructorName)
+            {
+                // Type '{0}' already defines a static constructor
+                diagnostics.Add(ErrorCode.ERR_StaticConstructorAlreadyExists, method1.Locations[0], this);
```

Comment:
Are we testing this case?

Reply:
@cston I'm not sure. Waiting for the build to see if there are build failures related to this or not. If there are not, I'll add new test for this.